File: src/org/newtonpolyhedron/solverprinters/MatrixMinorGCDSolverPrinter.java
package org.newtonpolyhedron.solverprinters;

import java.io.PrintWriter;
import java.math.BigInteger;
import java.util.List;

import org.apache.commons.lang3.tuple.Pair;
import org.apache.commons.math3.fraction.BigFraction;
import org.apache.commons.math3.linear.FieldMatrix;
import org.newtonpolyhedron.entity.SolverPrinter;
import org.newtonpolyhedron.solve.matrixminorgcd.MatrixMinorGCDSolver;

public class MatrixMinorGCDSolverPrinter extends SolverPrinter <MatrixMinorGCDSolver> {
	
	private final FieldMatrix <BigFraction>	baseMatrix;
	
	public MatrixMinorGCDSolverPrinter(
			final MatrixMinorGCDSolver solver,
			final FieldMatrix <BigFraction> baseMatrix,
			final PrintWriter output) {
		super(solver, output);
		this.baseMatrix = baseMatrix;
	}
	
	@Override
	public void solveFor(final MatrixMinorGCDSolver solver, final PrintWriter output) {
		final Pair <Integer, List <BigInteger>> result = solver.getLastRowGcd(baseMatrix);
		output.println(subheader("Last row minors:"));
		output.println(result.getRight());
		output.println("Minors GCD: ");
		output.println(result.getLeft());
	}
}

File: src/main/scala/org/newtonpolyhedron/solverprinters/MatrixMinorGCDSolverPrinter.scala
package org.newtonpolyhedron.solverprinters

import java.io.PrintWriter

import org.newtonpolyhedron.NewtonImports._
import org.newtonpolyhedron.entity.SolverPrinter
import org.newtonpolyhedron.solve.matrixminorgcd.MatrixMinorGCDSolver

class MatrixMinorGCDSolverPrinter[N <: MPNumber](
  override val solver: MatrixMinorGCDSolver[N],
  val baseMatrix:      Matrix[N],
  override val output: PrintWriter
)
    extends SolverPrinter[MatrixMinorGCDSolver[N]](solver, output) {

  override def solveFor(
      solver: MatrixMinorGCDSolver[N],
      output: PrintWriter
  ) = {
    output.println(title("Matrix GCD"))
    val result = solver.lastRowGcd(baseMatrix)
    output.println(subheader("Base matrix:"))
    output.println(baseMatrix)
    output.println("Last row minors:")
    output.println(result._2 mkString " ")
    output.println("Minors GCD: ")
    output.println(result._1)
  }
}
